### PR TITLE
Update StatsCommand Logic, Refactor Address to Cost

### LINKED ISSUE #38: As a user, I can calculate my total spending
so that I can keep track of my total spending.

### PR DESCRIPTION
**Pull request by choongyx**
#19 

Able to view total cost spent for all spendings (fixes #38)

### CODE REVIEW

File: src/test/java/seedu/address/model/spending/CostTest.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        String invalidCost = "";
+        assertThrows(IllegalArgumentException.class, () -> new Cost(invalidCost));
+    }
+
+    @Test
+    public void isValidCost() {
+        // null address
+        assertThrows(NullPointerException.class, () -> Cost.isValidCost(null));
+
+        // invalid addresses
```
Comment (Nanosync) [inline src/test/java/seedu/address/model/spending/CostTest.java]: invalid costs

File: src/test/java/seedu/address/model/spending/CostTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Test
+    public void constructor_invalidCost_throwsIllegalArgumentException() {
+        String invalidCost = "";
+        assertThrows(IllegalArgumentException.class, () -> new Cost(invalidCost));
+    }
+
+    @Test
+    public void isValidCost() {
+        // null address
```
Comment (Nanosync) [inline src/test/java/seedu/address/model/spending/CostTest.java]: null cost

File: src/test/java/seedu/address/model/spending/CostTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        assertThrows(NullPointerException.class, () -> Cost.isValidCost(null));
+
+        // invalid addresses
+        assertFalse(Cost.isValidCost("")); // empty string
+        assertFalse(Cost.isValidCost(" ")); // spaces only
+
+        // valid addresses
```
Comment (Nanosync) [inline src/test/java/seedu/address/model/spending/CostTest.java]: valid costs

**Review by Nanosync — APPROVED** [review]

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 4]
```diff
@@ -87,55 +87,55 @@ public void parse_compulsoryFieldMissing_failure() {
...
     @Test
     public void parse_invalidValue_failure() {
         // invalid name
-        assertParseFailure(parser, INVALID_NAME_DESC + DATE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+        assertParseFailure(parser, INVALID_NAME_DESC + DATE_DESC_BOB + EMAIL_DESC_BOB + COST_DESC_BOB
                 + TAG_DESC_HUSBAND + TAG_DESC_FRIEND, Name.MESSAGE_CONSTRAINTS);
 
-        // invalid date
-        assertParseFailure(parser, NAME_DESC_BOB + INVALID_DATE_DESC + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+        // invalid phone
```
Comment (austinsantoso) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: why is the comment being overwriten

File: src/test/java/seedu/address/logic/commands/EditSpendingDescriptorTest.java [hunk 5]
```diff
@@ -48,7 +48,7 @@ public void equals() {
         assertFalse(DESC_AMY.equals(editedAmy));
 
         // different address -> returns false
-        editedAmy = new EditSpendingDescriptorBuilder(DESC_AMY).withAddress(VALID_ADDRESS_BOB).build();
+        editedAmy = new EditSpendingDescriptorBuilder(DESC_AMY).withAddress(VALID_COST_BOB).build();
```
Comment (austinsantoso) [inline src/test/java/seedu/address/logic/commands/EditSpendingDescriptorTest.java]: why is the method name still withAddress?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 6]
```diff
@@ -49,36 +49,36 @@ public void parse_allFieldsPresent_success() {
...
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedSpending));
+                + COST_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedSpending));
 
         // multiple names - last name accepted
         assertParseSuccess(parser, NAME_DESC_AMY + NAME_DESC_BOB + DATE_DESC_BOB + EMAIL_DESC_BOB
-                + ADDRESS_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedSpending));
+                + COST_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedSpending));
 
-        // multiple dates - last date accepted
+        // multiple phones - last phone accepted
```
Comment (austinsantoso) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: shouldn't the comment be multiple dates

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 7]
```diff
@@ -87,55 +87,55 @@ public void parse_compulsoryFieldMissing_failure() {
...
+        // invalid phone
+        assertParseFailure(parser, NAME_DESC_BOB + INVALID_DATE_DESC + EMAIL_DESC_BOB + COST_DESC_BOB
                 + TAG_DESC_HUSBAND + TAG_DESC_FRIEND, Date.MESSAGE_CONSTRAINTS);
 
         // invalid email
-        assertParseFailure(parser, NAME_DESC_BOB + DATE_DESC_BOB + INVALID_EMAIL_DESC + ADDRESS_DESC_BOB
+        assertParseFailure(parser, NAME_DESC_BOB + DATE_DESC_BOB + INVALID_EMAIL_DESC + COST_DESC_BOB
                 + TAG_DESC_HUSBAND + TAG_DESC_FRIEND, Email.MESSAGE_CONSTRAINTS);
 
         // invalid address
```
Comment (austinsantoso) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: comment didn't change along

**Review by Nanosync — APPROVED** [review]

**Pull request merged**